contracts/FeeCustody.sol

```diff
@@ -0,0 +1,405 @@
+                _swap(asset, amountIn, _minAmountOut[i], _deadline);
+            }
+
+            // Transfer multisig allocation of protocol revenue to multisig
+            asset.transfer(protocolRevenueRecipient, multiSigRevenue);
+        }
+
+        toDistribute += distributionToken.balanceOf(address(this));
+        distributionToken.safeApprove(address(feeDistributor), toDistribute);
+
+        // Tranfer RBN locker allocation of protocol revenue to fee distributor
+        feeDistributor.burn(address(distributionToken), toDistribute);
+    }
+
+    /**
+     * @notice
+     * Amount of _asset allocated to RBN lockers from current balance
+     * @return amount allocated to RBN lockers
+     */
+    function claimableByRBNLockersOfAsset(address _asset)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 allocPCT = pctAllocationForRBNLockers;
+        return
+            IERC20(_asset).balanceOf(address(this)).mul(allocPCT).div(
+                TOTAL_PCT
+            );
+    }
+
+    /**
+     * @notice
+     * Amount of _asset allocated to multisig from current balance
+     * @return amount allocated to multisig
+     */
+    function claimableByProtocolOfAsset(address _asset)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+        return
+            IERC20(_asset).balanceOf(address(this)).mul(allocPCT).div(
+                TOTAL_PCT
+            );
+    }
+
+    /**
+     * @notice
+     * Total allocated to RBN lockers across all assets balances
+     * @return total allocated (in USD)
+     */
+    function totalClaimableByRBNLockersInUSD() external view returns (uint256) {
+        uint256 allocPCT = pctAllocationForRBNLockers;
+        return _getSwapQuote(allocPCT);
+    }
+
+    /**
+     * @notice
+     * Total allocated to multisig across all assets balances
+     * @return total allocated (in USD)
+     */
+    function totalClaimableByProtocolInUSD() external view returns (uint256) {
+        uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+        return _getSwapQuote(allocPCT);
+    }
+
+    /**
+     * @notice
+     * Total claimable across all asset balances based on allocation PCT
+     * @param _allocPCT allocation percentage
+     * @return total claimable (in USD)
+     */
+    function _getSwapQuote(uint256 _allocPCT)
+        internal
+        view
+        returns (uint256 claimable)
+    {
+        for (uint256 i; i < lastAssetIdx; i++) {
+            IChainlink oracle = IChainlink(oracles[assets[i]]);
+
+            // Approximate claimable by multiplying
+            // current asset balance with current asset price in USD
+            claimable += IERC20(assets[i])
+                .balanceOf(address(this))
+                .mul(oracle.latestAnswer())
+                .mul(_allocPCT)
+                .div(oracle.decimals())
```

Comment:
All the USD oracle's `decimals()` should return 8, so shall we just hardcode this to 8?

Reply:
This is because each `decimals` call will take up 20k gas, quite expensive if called from a contract